Source/NSButton.m

```diff
@@ -543,4 +547,43 @@ - (NSSound *) sound
   return [_cell sound];
 }
 
+- (void) mouseDown: (NSEvent *)event
```

Comment:
Are you sure this behaviour is only required for mouse interaction and not for keyboard triggered state changes?

Reply:
I believe it's needed for all.  I will find a way to do it in one location.

Reply:
One option could be to call this logic from within the setState: method of the cell. The implementation would still need to be on the controller, that is here within NSButton though.

Reply:
I moved it to sendAction:toTarget: which I believe is central enough to be called in all cases.   Let me know if you consider this a good place for it as well.

Reply:
I believe this should now be resolved.  I have moved it to a location in the code where it is called in all cases.